### PR TITLE
tools/c7n-org - account id validation

### PR DESCRIPTION
**Pull request by kapilt**
validate aws account id conforms to expected value.

closes https://github.com/cloud-custodian/cloud-custodian/issues/4545

### CODE REVIEW

File: tools/c7n_org/c7n_org/cli.py [hunk 1]
```diff
@@ -74,7 +74,10 @@
             'properties': {
                 'name': {'type': 'string'},
                 'email': {'type': 'string'},
-                'account_id': {'type': 'string'},
+                'account_id': {
+                    'type': 'string',
+                    'pattern': '^[0-9]{12}$',
```
Comment (ingwarsw) [inline tools/c7n_org/c7n_org/cli.py]: Its only for AWS?
It will not break GCP or azure?

Comment (kapilt) [reply tools/c7n_org/c7n_org/cli.py]: we use gcp project and azure subscription respectively for those in accounts.yml

**Comment by kapilt** [comment]
azure sdk change breaking ci again

File: tools/c7n_org/c7n_org/cli.py (hunk 1 above)
Comment (kapilt) [reply tools/c7n_org/c7n_org/cli.py]: added some basic account config validation tests across providers with required values.

File: tools/c7n_org/tests/test_org.py [hunk 2]
```diff
@@ -60,6 +75,42 @@ def setup_run_dir(self, accounts=None, policies=None):
...
+        self.change_cwd(run_dir)
+        runner = CliRunner()
+        result = runner.invoke(
+            org.cli,
+            ['run', '-c', 'accounts.yml', '-u', 'policies.yml',
+             '--debug', '-s', 'output', '--cache-path', 'cache'],
+            catch_exceptions=False)
+        self.assertEqual(result.exit_code, 0)
+
+    def test_validate_gcp_provider(self):
```
Comment (kapilt) [inline tools/c7n_org/tests/test_org.py]: this should really be parameterized with pytest

Comment (kapilt) [reply tools/c7n_org/tests/test_org.py]: switching to pytest parameterize actually results in a significant diff increase as we need to switch out from our existing usage of unittest classes to get pytests parameterize support.

**Pull request merged**